packages/sol-compiler/src/utils/compiler.ts

```diff
@@ -232,9 +232,13 @@ export function getSourceTreeHash(resolver: Resolver, importPath: string): Buffe
     if (dependencies.length === 0) {
         return sourceHash;
     } else {
-        const dependencySourceTreeHashes = _.map(dependencies, (dependency: string) =>
-            getSourceTreeHash(resolver, dependency),
-        );
+        const dependencySourceTreeHashes = _.map(dependencies, (dependency: string) => {
+            try {
+                return getSourceTreeHash(resolver, dependency);
+            } catch (e) {
+                throw Error(`Error when trying to resolve dependencies for ${importPath}: ${(e as Error).message}`);
```

Comment:
This error will get pretty weird whenever the dependency resolution fails in a dependency of a contract. For example, if an import path is altered in `contracts/exchange/src/MixinTransactions.sol`, the error message will look something like this when compiling `Exchange.sol`:

```
Error: Error when trying to resolve dependencies for src/Exchange.sol: Error when trying to resolve dependencies for src/MixinMatchOrders.sol: Error when trying to resolve dependencies for src/MixinExchangeCore.sol: Error when trying to resolve dependencies for src/MixinSignatureValidator.sol: Error when trying to resolve dependencies for src/MixinTransactions.sol: Failed to resolve src/interfaces/Transactions.sol
    at /Users/alextowle/0x-monorepo-1/packages/sol-compiler/src/utils/compiler.ts:239:23
    at arrayMap (/Users/alextowle/0x-monorepo-1/node_modules/lodash/lodash.js:639:23)
    at Function.map (/Users/alextowle/0x-monorepo-1/node_modules/lodash/lodash.js:9554:14)
    at Object.getSourceTreeHash (/Users/alextowle/0x-monorepo-1/packages/sol-compiler/src/utils/compiler.ts:235:46)
    at Compiler.<anonymous> (/Users/alextowle/0x-monorepo-1/packages/sol-compiler/src/compiler.ts:240:39)
    at step (/Users/alextowle/0x-monorepo-1/packages/sol-compiler/lib/src/compiler.js:43:23)
    at Object.next (/Users/alextowle/0x-monorepo-1/packages/sol-compiler/lib/src/compiler.js:24:53)
    at fulfilled (/Users/alextowle/0x-monorepo-1/packages/sol-compiler/lib/src/compiler.js:15:58)
    at processTicksAndRejections (internal/process/task_queues.js:86:5)
error Command failed with exit code 1.
```

This could be avoided by checking to see which kind of error is being thrown, and pass up errors of the form `Error when trying to resolve dependencies for $SOME_CONTRACT.sol: Failed to resolve $SOME_IMPORT_PATH.